cmd/kube-apiserver/app/server.go

```diff
@@ -239,6 +239,7 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	// Kubelet related flags:
 	fs.BoolVar(&s.KubeletConfig.EnableHttps, "kubelet-https", s.KubeletConfig.EnableHttps, "Use https for kubelet connections")
 	fs.UintVar(&s.KubeletConfig.Port, "kubelet-port", s.KubeletConfig.Port, "Kubelet port")
+	fs.MarkDeprecated("kubelet-port", "kubelet-port is not used anymore and it will be removed")
```

Comment:
This is misleading. It IS used, if the nodes don't report kubelet ports. It will not be used in the future.

This, btw, is a breaking change for all our partners, whose turnup code isn't in our repo, so ensure it's called out in the release notes.